File: SportsPlatform.Auth/SportsPlatform.Auth.Api/wwwroot/react_web/src/services/clubService.js
import api from './api';

// GET /clubs/my
export const getMyClubs = async () => {
  const { data } = await api.get('/clubs/my');
  return data;
};

// GET /clubs/:clubId
export const getClub = async (clubId) => {
  const { data } = await api.get(`/clubs/${clubId}`);
  return data;
};

// POST /clubs
export const createClub = async ({ name }) => {
  const { data } = await api.post('/clubs', { name });
  return data;
};

// DELETE /clubs/:clubId
export const deleteClub = async (clubId) => {
  const { data } = await api.delete(`/clubs/${clubId}`);
  return data;
};

// GET /clubs/:clubId/members
export const getClubMembers = async (clubId) => {
  const { data } = await api.get(`/clubs/${clubId}/members`);
  return data;
};

// DELETE /clubs/:clubId/members/:userId
export const removeClubMember = async (clubId, userId) => {
  const { data } = await api.delete(`/clubs/${clubId}/members/${userId}`);
  return data;
};

// POST /clubs/:clubId/invitations
export const createClubInvitation = async (clubId, inviteData) => {
  const { data } = await api.post(`/clubs/${clubId}/invitations`, inviteData);
  return data;
};

// GET /clubs/:clubId/invitations
export const getClubInvitations = async (clubId) => {
  const { data } = await api.get(`/clubs/${clubId}/invitations`);
  return data;
};

// DELETE /clubs/:clubId/invitations/:invitationId
export const cancelClubInvitation = async (clubId, invitationId) => {
  const { data } = await api.delete(`/clubs/${clubId}/invitations/${invitationId}`);
  return data;
};

File: eq-master/lib/services/club_service.dart
import 'package:flutter/foundation.dart';
import 'package:file_picker/file_picker.dart';

import '../models/api_models.dart';
import 'api_client.dart';

class ClubService {
  final ApiClient _api = ApiClient.instance;

  Future<List<ClubDto>> getMyClubs() async {
    final json = await _api.get('/clubs/my');
    return (json as List)
        .map((e) => ClubDto.fromJson(Map<String, dynamic>.from(e as Map)))
        .toList();
  }

  Future<void> leaveClub(String clubId, String currentUserId) {
    return _api.delete('/clubs/$clubId/members/$currentUserId');
  }

  Future<ClubDto> createClub(
    String name, {
    PlatformFile? logo,
    String? location,
    double? locationLatitude,
    double? locationLongitude,
  }) async {
    final body = {
      'name': name,
      if (location?.trim().isNotEmpty == true) 'location': location!.trim(),
      if (locationLatitude != null) 'locationLatitude': locationLatitude,
      if (locationLongitude != null) 'locationLongitude': locationLongitude,
    };
    final fields = {
      'name': name,
      if (location?.trim().isNotEmpty == true) 'location': location!.trim(),
      if (locationLatitude != null)
        'locationLatitude': locationLatitude.toString(),
      if (locationLongitude != null)
        'locationLongitude': locationLongitude.toString(),
    };
    final json = logo == null
        ? await _api.post('/clubs', body: body)
        : await _api.uploadFile(
            '/clubs',
            fileField: 'logo',
            fileBytes: logo.bytes,
            filePath: kIsWeb ? null : logo.path,
            fileName: logo.name,
            fields: fields,
          );
    return ClubDto.fromJson(Map<String, dynamic>.from(json as Map));
  }
}
